lib/rubocop/cop/root_cops/factories/factory_name.rb

```diff
@@ -0,0 +1,76 @@
+require "active_support/inflector"
+require_relative "../../../helpers/factories.rb"
+
+module RuboCop
+  module Cop
+    module RootCops
+      module Factories
+        class FactoryName < Cop
+          def investigate(processed_source)
+            file_path = processed_source.buffer.name
+            @base_file_name = File.basename(file_path, ".rb")
+            @base_file_name_body = _generate_body(@base_file_name)
+            @base_file_name_last_word = @base_file_name.split("_").last
+          end
+
+          def on_send(node)
+            return if _file_name_has_error?
+
+            _, method_name = *node
+
+            return unless _factory_method?(method_name)
+
+            factory_symbol_name = *node.arguments[0]
+            factory_name_array = factory_symbol_name[0].id2name.split("__")
+            factory_name = factory_name_array.first
+            factory_name_body = _generate_body(factory_name)
+            factory_name_last_word = factory_name.split("_").last
+
+            if factory_name_array.size > 1
+              if factory_name_array[0] != @base_file_name
+                add_offense(
+                  node,
+                  :location => :expression,
+                  :message => "Factory name uses incorrect prefix, should be '#{@base_file_name}__#{factory_name_array[1]}'."
+                )
+              end
+
+            elsif factory_name == @base_file_name && _not_invariant_noun?(@base_file_name_last_word)
+              add_offense(
+                node,
+                :location => :expression,
+                :message => "Factory name should be singular form of the file name."
+              )
+
+            elsif (factory_name_body != @base_file_name_body) || (factory_name_last_word.pluralize != @base_file_name_last_word)
+              add_offense(
+                node,
+                :location => :expression,
+                :message => "Factory should be in own file or the singular form of the file name. If you want closely related factories in the same file then name the factory '#{@base_file_name}__#{factory_name}'."
+              )
+            end
+          end
+
+          private
+
+          def _factory_method?(method)
+            method == :factory
+          end
+
+          def _not_invariant_noun?(word)
+            word.pluralize != word.singularize
```

Comment:
Did you have a chance to make sure root-server's inflection configurations are used with this?

Reply:
Oops! No I still need to do that. Do you think the cop should be checking every project to make sure it has an inflection configuration file, or just root-server? For example, do we also want rooter-bot-rails to be using an one, too?

Reply:
I don't think you need to check if the project has one. It's up to each project if they need one. Instead, I was wondering if the cop uses the project's config when running (i.e. run the cop and make sure it doesn't break for `data`)

Reply:
Ah I get what you mean now. I just ran the two cops on root-server, and it looks like FactoryName cop doesn't break for `data`, but the FactoryFileName cop complains :/

For example, `factories/driving_incident_rating_data` passes the FactoryName cop, but we get this complaint: 
```
factories/driving_incident_rating_data.rb C: Factories/FactoryName: Factory name should be singular form of the file name.
  factory :driving_incident_rating_data do
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I will look into why this is failing

Reply:
It might not be loading the config since rubocop probably isn't loading rails. I'm not entirely sure what the best approach would be to fix that, though, since duplicating our inflections in two places doesn't seem like the best